controllers/feeds/getFeedChat.js

```diff
@@ -148,12 +148,18 @@ const finalize = (req, id, myReaction, newGroup, activity_id, constantActor, kar
       (data) => data.reaction.kind === 'comment'
     );
     if (listComment.length > 0) {
-      newGroup[activity_id].data.last_message_at = listComment?.sort(
-        (a, b) => new Date(a.reaction.created_at) - new Date(b.reaction.created_at)
-      )?.[0]?.reaction?.created_at;
-      newGroup[activity_id].data.updated_at = listComment?.sort(
-        (a, b) => new Date(a.reaction.updated_at) - new Date(b.reaction.updated_at)
-      )?.[0]?.reaction?.created_at;
+      const sortedCommentByCreatedAt = listComment.sort(
+        (a, b) => new Date(b.reaction.created_at) - new Date(a.reaction.created_at)
+      );
+
+      const sortedCommentByUpdatedAt = listComment.sort(
+        (a, b) => new Date(b.reaction.updated_at) - new Date(a.reaction.updated_at)
+      );
+
+      newGroup[activity_id].data.last_message_at =
+        sortedCommentByCreatedAt?.[0]?.reaction?.created_at;
+
+      newGroup[activity_id].data.updated_at = sortedCommentByUpdatedAt?.[0]?.reaction?.created_at;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes look good, but there's a potential issue here. If `sortedCommentByCreatedAt` or `sortedCommentByUpdatedAt` are empty arrays, trying to access the `reaction` property on the first element will result in a TypeError because you're trying to access a property of `undefined`. To prevent this, you should check if the first element exists before trying to access its properties.

```diff
- newGroup[activity_id].data.last_message_at =
-   sortedCommentByCreatedAt?.[0]?.reaction?.created_at;
- 
- newGroup[activity_id].data.updated_at = sortedCommentByUpdatedAt?.[0]?.reaction?.created_at;

+ newGroup[activity_id].data.last_message_at =
+   sortedCommentByCreatedAt?.[0] ? sortedCommentByCreatedAt[0].reaction.created_at : null;
+ 
+ newGroup[activity_id].data.updated_at = sortedCommentByUpdatedAt?.[0] ? sortedCommentByUpdatedAt[0].reaction.updated_at : null;
```
This way, if the first element doesn't exist, `last_message_at` and `updated_at` will be set to `null` instead of throwing an error.


<!-- This is an auto-generated comment by OSS CodeRabbit -->